src/content/docs/en/reference/dev-toolbar-app-reference.mdx

```diff
@@ -2,278 +2,392 @@
-        detail: {
-          state: true,
-        },
-      })
-    );
-  }
-}
-```
+### `addDevToolbarApp()`
 
+A function available to [the `astro:config:setup` hook](/en/reference/integrations-reference/#astroconfigsetup) that adds dev toolbar apps. It takes an object with the following required properties to define the toolbar app: [`id`](#id), [`name`](#name), [`icon`](#icon), and [`entrypoint`](#entrypoint).
 
-### `id: string`
+### `id`
 
 A unique identifier for the app. This will be used to uniquely identify the app in hooks and events.
 
-```ts title="src/my-app.js" {2}
-export default {
+```ts title="my-integration.js" {2}
+{
   id: 'my-app',
   // ...
 }
 ```
 
-### `name: string`
+### `name`
 
 The name of the app. This will be shown to users whenever the app needs to be referenced using a human-readable name.
 
-```ts title="src/my-app.js" {3}
-export default {
+```ts title="my-integration.js" {3}
+{
   // ...
   name: 'My App',
   // ...
 }
 ```
 
-### `icon: Icon`
+### `icon`
 
-The icon of the app. This will be used to display the app in the UI. This can either be an icon from [the icon list](#icons), or a string containing the SVG markup of the icon.
+The icon used to display the app in the toolbar. This can either be an icon from [the icon list](#icons), or a string containing the SVG markup of the icon.
 
-```ts title="src/my-app.js" {3}
-export default {
+```ts title="my-integration.js" {3}
+{
   // ...
-  icon: '<svg>...</svg>', // or 'astro:logo'
+  icon: '<svg>...</svg>', // or, e.g. 'astro:logo'
   // ...
 }
 ```
 
-### `init: (canvas: ShadowRoot, eventTarget: EventTarget) => void`
+### `entrypoint`
+
+The path to the file that exports the dev toolbar app.
+
+```ts title="my-integration.js" {3}
+{
+  // ...
+  entrypoint: './my-app.js',
+}
+```
+
+## Structure of a Dev Toolbar App
+
+A Dev Toolbar App is a `.js` or `.ts` file that default exports an object using the [`defineToolbarApp()` function](#definetoolbarapp) available from the `astro/toolbar` module. 
+
+```ts title="src/my-app.js"
+import { defineToolbarApp } from "astro/toolbar";
+
+export default defineToolbarApp({
+  init(canvas) {
+    const text = document.createTextNode('Hello World!');
+    canvas.appendChild(text);
+  }
+  beforeTogglingOff() {
+    const confirmation = window.confirm('Really exit?');
+    return confirmation;
+  } 
+});
+```
+### `defineToolbarApp()`
+
+<p><Since v="4.7.0" /></p>
+
+A function that defines the logic of your toolbar app when it is loaded and toggled off. 
 
-This is the core part of the app. This function will be called when the app is loaded, which will either be when the browser is idle or when the user clicks on the app in the UI.
+This function takes an object with an [`init()`](#init) function that will be called when the dev toolbar app is loaded. It can also take a [`beforeTogglingOff()`](#beforetogglingoff) function that will run when the toolbar app is clicked to toggle off the its active status.
```

Comment:
«...toggle off the it's active status...» or
«...toggle off its active status...»

Reply:
Another great catch, thank you!